### PR TITLE
Add Supermicro ZTP policies

### PR DESCRIPTION
**Pull request by natifridman**
- Separate common policies from cluster policies
- Add Goldwing cluster policies
- Fix ztp-policies appset project

### CODE REVIEW

**Review by nocturnalastro — APPROVED** [review]

**Review by novacain1 — CHANGES_REQUESTED** [review]

File: clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          reserved: "0-1,32-33"
+        hugepages:
+          defaultHugepagesSize: "1G"
+          pages:
+            - size: "1G"
+              count: 64
+              node: 0
+            - size: "1G"
+              count: 64
+              node: 1
```
[the remark this answers is not in the document]
Comment (novacain1) [reply clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml:33]: Delete node:1 and its corresponding attributes.

File: clusters/ztp-siteconfig/goldwing/goldwing-siteconfig.yaml [hunk 2]
```diff
@@ -14,7 +14,7 @@ spec:
   clusters:
     - clusterName: "goldwing"
       clusterLabels:
-        group-vdu: ""
```
[the remark this answers is not in the document]
Comment (natifridman) [reply clusters/ztp-siteconfig/goldwing/goldwing-siteconfig.yaml:17]: > I would've thought that group-vdu would be enabled here for Goldwing, as this PGT contains Operators to install specifically for a vDU. Given discussion today though, we will need to augment that for a different PTP operator to satisfy deficiencies in functionality that exist in 4.12's version that lacks GPSd.

I thought that `common: "true"` is the one with the operators installation

File: clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (natifridman) [reply clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml]: done

File: clusters/ztp-siteconfig/goldwing/goldwing-siteconfig.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (novacain1) [reply clusters/ztp-siteconfig/goldwing/goldwing-siteconfig.yaml:17]: you are correct on common

File: clusters/ztp-policies/common/group-policies/group-supermicro.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    - fileName: PerformanceProfile.yaml
+      policyName: "config-policy"
+      spec:
+        additionalKernelArgs:
+          - crashkernel=512M
+          # - cpufreq.default_governor=schedutil 30-Jan testing dcain
+          - rcupdate.rcu_normal_after_boot=0
+        cpu:
+          # These must be tailored for the specific hardware platform
+          isolated: "2-31, 34-63"
```
Comment (novacain1) [inline clusters/ztp-policies/common/group-policies/group-supermicro.yaml:34]: missing reserved

Comment (novacain1) [reply clusters/ztp-policies/common/group-policies/group-supermicro.yaml:34]: Do you aim to have the PerformnaceProfile in the group policies or in site specific?  you should stick to one place.  i would advise you put it in a group specific policy that matches that class of hardware that I ordered (SMC X11) in your case for Goldwing.

File: clusters/ztp-policies/goldwing/cluster-specific-policies/goldwing-cluster-specific.yaml
Comment (novacain1) [inline clusters/ztp-policies/goldwing/cluster-specific-policies/goldwing-cluster-specific.yaml:1]: I might be confused, but I thought the intention of Goldwing would be to use an early PTP operator for testing from Jose, in which case I would not use the DU profile's configuration of a T-SC.

File: clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  mcp: "master"
+  sourceFiles:
+    - fileName: PerformanceProfile.yaml
+      policyName: "perfprofile-policy"
+      metadata:
+        name: openshift-node-performance-profile
+      spec:
+        additionalKernelArgs:
+          - rcupdate.rcu_normal_after_boot=0
+          - nohz=off
```
Comment (novacain1) [inline clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml:20]: remove this

File: clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          - nohz_full=2-31, 34-63
```
Comment (novacain1) [inline clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml:21]: remove this

**Review by novacain1 — CHANGES_REQUESTED** [review]
See comments, almost there

File: clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml (hunk 4 above)
Comment (natifridman) [reply clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml]: done

File: clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml (hunk 5 above)
Comment (natifridman) [reply clusters/ztp-policies/goldwing/site-specific-policies/goldwing-site-specific.yaml]: done

File: clusters/ztp-policies/goldwing/cluster-specific-policies/goldwing-cluster-specific.yaml
Comment (natifridman) [reply clusters/ztp-policies/goldwing/cluster-specific-policies/goldwing-cluster-specific.yaml:1]: That's why I didn't apply it [here](https://github.com/redhat-partner-solutions/vse-carslab-hub/blob/83edbb0e8ea317c172ce1021eacea03ffea08321/clusters/ztp-policies/goldwing/kustomization.yaml#L7)

File: clusters/ztp-policies/common/group-policies/group-supermicro.yaml (hunk 3 above)
Comment (natifridman) [reply clusters/ztp-policies/common/group-policies/group-supermicro.yaml:34]: This is something I wasn't sure about. Should I remove just the `cpu` from the site specific or something more?

**Review by hhamid93 — APPROVED** [review]

File: clusters/ztp-policies/common/group-policies/group-supermicro.yaml (hunk 3 above)
Comment (novacain1) [reply clusters/ztp-policies/common/group-policies/group-supermicro.yaml:34]: I think you should remove goldwing-site-specific.yaml entirely, based on looking at group-supermicro.yaml you are duplicating this content.

File: clusters/ztp-policies/goldwing/cluster-specific-policies/goldwing-cluster-specific.yaml
Comment (natifridman) [reply clusters/ztp-policies/goldwing/cluster-specific-policies/goldwing-cluster-specific.yaml:1]: Removed this file, will add it back if needed

**Review by novacain1 — APPROVED** [review]
LGTM

**Pull request merged**